inference/src/basics.rs

```diff
@@ -377,27 +453,35 @@ impl FOModule {
             solver.assert(&Term::negate(t.clone()));
 
             let resp = solver.check_sat(HashMap::new()).expect("error in solver");
-            match resp {
-                SatResp::Sat => {
-                    let mut states = solver
-                        .get_minimal_model()
-                        .expect("solver error while minimizing");
-                    assert_eq!(states.len(), 1);
-
-                    if !core.add_counter_model(states[0].clone(), None) {
-                        solver.save_tee();
-                        return Some(states.pop().unwrap());
+            let cex_res = match resp {
+                SatResp::Sat => get_minimal_model(&mut solver),
+                SatResp::Unsat => get_unsat_core(&mut solver),
+                SatResp::Unknown(reason) => CexResult::Unknown(reason),
+            };
+
+            solver.save_tee();
+
+            cex_res
+        };
+
+        let mut conf_idx = 0;
+        let mut unknown_reasons = vec![];
+        while conf_idx < confs.len() {
+            match try_conf(confs[conf_idx], &mut core) {
+                CexResult::Cex(model) => {
+                    if !core.add_counter_model(model[0].clone()) {
+                        return CexResult::Cex(model);
```

Comment:
Can you add a comment here about the control flow? There also isn't a comment on `add_counter_model` explaining what the boolean return value is.